site-src/geps/gep-1757.md

```diff
@@ -0,0 +1,224 @@
+
+The spec currently specifies that gateways *can* be merged. This MUST be implemented by referencing the same "existing infrastructure" (as defined in "Manual Deployments").
+
+For example:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: parent-gateway
+spec:
+  gatewayClassName: example
+  listeners:
+  - name: default
+    port: 80
+    protocol: HTTP
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: merged-gateway
+spec:
+  addresses:
+  - type: NamedAddress
+    value: parent-gateway
+  gatewayClassName: example
+  listeners:
+  - name: some-other-route
+    port: 81
+    protocol: HTTP
+```
+
+With this configuration, an implementation:
+* MUST treat the `parent-gateway` as having the concatenated list of all listeners from merged Gateways.
+  This means, for example, that a Service with port 80 and 81 may be created in this example.
+
+### Customizations
+
+With any in-cluster deployment, customization requirements will arise. 
+
+Some common requirements would be:
+* `Service.spec.type`, to control whether a service is a `ClusterIP` or `LoadBalancer`.
+* If it is a `LoadBalancer`, the `Service.spec.loadBalancerIP` to assign to it.
+* Arbitrary labels and annotations on generated resources.
+* CPU and memory requests, and other similar customizations.
+
+For setting a specific IP address in the Service, the `.Spec.Address` field can be used. For example:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: my-gateway
+spec:
+  addresses:
+  - type: IPAddress
+    value: 1.1.1.1
+  gatewayClassName: example
+  listeners:
+  - name: default
+    port: 80
+    protocol: HTTP
+```
+
+This differs from the selection of "manual deployment", which doesn't use `IPAddress` type.
+
+Currently, all other customizations are implementation specific.
+Temporary Note: it seems like most implementations are using `parametersRef` in GatewayClass. The author thinks this is a bit problematic.
+
+### Resource Attachment
+
+Generated resources are guaranteed to have two attributes:
+* A `gateway.networking.k8s.io/metadata.name: <NAME>` label.
+* A name `<NAME>-<GATEWAY CLASS>`.
+
+These can be relied on to attach resources to.
+While "Policy attachment" in the gateway-api would use attachment to the actual `Gateway` resource itself,
+many existing things attach only to resources like `Deployment` or `Service`.
+
+An example using these:
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: gateway
+spec:
+  gatewayClassName: example
+  listeners:
+  - name: default
+    hostname: "example.com"
+    port: 80
+    protocol: HTTP
+---
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: gateway
+spec:
+  # Match the generated Deployment by reference
+  # Note: Do not use `kind: Gateway`.
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: gateway-example
```

Comment:
Maybe we could introduce a simple mutating webhook that would translate Gateway references to this format?

Reply:
Maybe. Could be a bit confusing though when user looks at it (also its stateful since you need to lookup the gateway class name).

IMO making Gateway implement /scale would be pretty neat